rs/types/management_canister_types/src/lib.rs

```diff
@@ -998,6 +1000,22 @@ pub struct CanisterStatusResultV2 {
     query_stats: QueryStats,
 }
 
+#[derive(Eq, PartialEq, Debug, CandidType, Deserialize)]
+pub struct ExecutionMemorySize {
+    wasm_memory_size: candid::Nat,
+    stable_memory_size: candid::Nat,
+    global_memory_size: candid::Nat,
+    wasm_binary_memory_size: candid::Nat,
+    custom_sections_memory_size: candid::Nat,
+}
+
+#[derive(Eq, PartialEq, Debug, CandidType, Deserialize)]
+pub struct SystemMemorySize{
+    canister_history_memory_size: candid::Nat,
+    wasm_chunk_store_memory_size: candid::Nat,
+    snapshot_memory_size: candid::Nat,
+}
```

Comment:
Suggestion:

```suggestion
#[derive(Eq, PartialEq, Debug, CandidType, Deserialize)]
pub struct ExecutionMemorySize {
    heap: candid::Nat,
    stable: candid::Nat,
    globals: candid::Nat,
    wasm_binary: candid::Nat,
    custom_sections: candid::Nat,
}

#[derive(Eq, PartialEq, Debug, CandidType, Deserialize)]
pub struct SystemMemorySize{
    canister_history: candid::Nat,
    wasm_chunk_store: candid::Nat,
    snapshot: candid::Nat,
}
```

Reply:
See above on combining both of these into a single record/struct.

If all of these are inside a record named `memory_metrics`, maybe we just suffix each of these with `size`.